src/lib/integration_test_cloud_engine/mina_automation.ml

```diff
@@ -376,65 +379,73 @@ module Network_manager = struct
         |> Terraform.to_string
         |> Out_channel.output_string ch) ;
     let testnet_log_filter = Network_config.testnet_log_filter network_config in
-    let cons_node pod_id mina_container_id mina_archive_container_id
-        network_keypair_opt =
-      { Kubernetes_network.Node.testnet_name =
-          network_config.terraform.testnet_name
-      ; cluster = cluster_id
-      ; namespace = network_config.terraform.testnet_name
-      ; pod_id
-      ; mina_container_id
-      ; mina_archive_container_id
-      ; graphql_enabled = network_config.terraform.deploy_graphql_ingress
-      ; network_keypair = network_keypair_opt
-      }
+    let cons_workload workload_id node_info : Kubernetes_network.Workload.t =
+      { workload_id; node_info }
+    in
+    let cons_node_info ?network_keypair ?(has_archive_container = false)
+        primary_container_id : Kubernetes_network.Node.info =
+      { network_keypair; has_archive_container; primary_container_id }
     in
     (* we currently only deploy 1 seed and coordinator per deploy (will be configurable later) *)
-    let seed_nodes = [ cons_node "seed" "mina" None None ] in
-    let snark_coordinator_name =
-      "snark-coordinator-"
-      ^ String.lowercase
-          (String.sub network_config.terraform.snark_worker_public_key
-             ~pos:
-               ( String.length network_config.terraform.snark_worker_public_key
-               - 6 )
-             ~len:6)
-    in
-    let snark_coordinator_nodes =
+    let seed_workloads = [ cons_workload "seed" [ cons_node_info "mina" ] ] in
+    let snark_coordinator_id =
+      String.lowercase
+        (String.sub network_config.terraform.snark_worker_public_key
+           ~pos:
+             (String.length network_config.terraform.snark_worker_public_key - 6)
+           ~len:6)
+    in
+    let snark_coordinator_workloads =
       if network_config.terraform.snark_worker_replicas > 0 then
-        [ cons_node snark_coordinator_name "coordinator" None None ]
+        [ cons_workload
+            ("snark-coordinator-" ^ snark_coordinator_id)
+            [ cons_node_info "mina" ]
+        ]
       else []
     in
-    let block_producer_nodes =
+    let snark_worker_workloads =
```

Comment:
This is also only if `network_config.terraform.snark_worker_replicas > 0 `, right? otherwise it creates an empty list of nodes under the snark worker workload which wouldn't exist either (the reason why intg tests are failing)

Reply:
good catch, fixing this